### PR TITLE
#1001 Added script to find a python executable, and use it for all calls to process.py

### PR DESCRIPTION
**Pull request by hiker**
As discussed in #1001 . I decided not to try to install process.py (under whatever name), since I think it's better not to 'pollute' the user's PATH with unnecessary binaries. Once we install the psydata libraries (i.e. the lib directory), all makefiles use relative references to the scripts, so they all should work without any change.

### CODE REVIEW

File: lib/get_python.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/bash
```
Comment (arporter) [inline lib/get_python.sh:1]: Please add copyright/license info.

File: lib/get_python.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# This scripts find an executable python command. On
```
Comment (arporter) [inline lib/get_python.sh:3]: "This script finds an..."

File: lib/get_python.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    # Necessary in case that 'python' is a directory
+    if [[ -f "$p" && -x $(realpath "$p") ]]; then
+        echo $p;
+        exit
+    fi
+done
+
+echo "Cannot find working python"
+
+exit -1
```
Comment (arporter) [inline lib/get_python.sh:24]: Missing new-line.

**Review by arporter — CHANGES_REQUESTED** [review]
All Makefiles under lib have been changed.
Fix is straightforward and makes sense.
This branch works OK for me on a WSL2 installation (which only has python3).
I have no Mac to test this on so, for completeness, please could either @deardenchris or @christophermaynard checkout this branch on their Mac, `cd lib/nan_test` and do `make` and check that there are no errors.
Apart from that that there are just a few minor typos and a missing license/copyright.

**Comment by deardenchris** [comment]
Hi Andy, I've just tried this on my Mac. I found that I had to change `#!/usr/bin/bash` in `get_python.sh` to `#!/bin/bash` otherwise I got the error `/bin/sh: ../get_python.sh: /usr/bin/bash: bad interpreter: No such file or directory`. Although regardless of whether I make the change or not to `get_python.sh`, the same output is produced in the nan_test directory (i.e. the same .mod files and object files) so I think the build is OK?

**Comment by hiker** [comment]
At least on ubuntu there is also ``/bin/bash``, so that might work for everything? Alternatively, google indicates that ``#!/usr/bin/env bash`` might be a portable option (https://forums.fedoraforum.org/showthread.php?302429-BASH-vs-SHELL-usr-bin-bash-vs-bin-bash&s=f6d965a4a5d5cc63b68910950adbbbcd&p=1720154#post1720154) ... but it also mentions that 'nearly always' ``env`` is ``/usr/bin/env``

So, since on all platforms it seems that `/bin/bash`` works, I'll just use this.

File: lib/get_python.sh (hunk 1 above)
Comment (hiker) [reply lib/get_python.sh]: Done.

File: lib/get_python.sh (hunk 2 above)
Comment (hiker) [reply lib/get_python.sh]: Oops.

File: lib/get_python.sh (hunk 3 above)
Comment (hiker) [reply lib/get_python.sh]: Done.

**Comment by hiker** [comment]
I've changed it to ``/bin/bash``, which seems to work on nearly all systems (I tried various systems at BOM, and all had ``/bin/bash``). Ready for next review.

File: lib/get_python.sh [hunk 4]
```diff
@@ -1,6 +1,42 @@
-#!/usr/bin/bash
+#!/usr/bash
```
Comment (arporter) [inline lib/get_python.sh:1]: Should this be `/bin/bash`?

**Review by arporter — CHANGES_REQUESTED** [review]
All requested changes have been made but the script now refers to `/usr/bash` when I think it should be `/bin/bash`?

**Review by arporter — APPROVED** [review]
All good now.

**Pull request merged**